operator/src/main/java/oracle/kubernetes/operator/rest/WebhookRestServer.java

```diff
@@ -0,0 +1,319 @@
+ *
+ */
+public class WebhookRestServer {
+  private static final LoggingFacade LOGGER = LoggingFactory.getLogger("Webhook", "Operator");
+  private static final int CORE_POOL_SIZE = 2;
+  private static final String SSL_PROTOCOL = "TLSv1.2";
+  private static final String[] SSL_PROTOCOLS = {
+    SSL_PROTOCOL
+  }; // ONLY support TLSv1.2 (by default, we would get TLSv1 and TLSv1.1 too)
+  private static WebhookRestServer INSTANCE = null;
+  private final WebhookRestConfig config;
+  // private String baseHttpUri;
+  private final String baseHttpsUri;
+  private HttpServer httpsServer;
+
+  /**
+   * Constructs the conversion webhook REST server.
+   *
+   * @param config - contains the webhook REST server's configuration, which includes the hostnames and port
+   *     numbers that the ports run on, the certificates and private keys for ssl, and the backend
+   *     implementation that does the real work behind the REST api.
+   */
+  private WebhookRestServer(WebhookRestConfig config) {
+    LOGGER.entering();
+    this.config = config;
+    baseHttpsUri = "https://" + config.getHost() + ":" + config.getHttpsPort();
+    LOGGER.exiting();
+  }
+
+  /**
+   * Create singleton instance of the conversion webhook's RestServer. Should only be called once.
+   *
+   * @param restConfig - the conversion webhook's REST configuration. Throws IllegalStateException if
+   *     instance already created.
+   */
+  public static synchronized void create(WebhookRestConfig restConfig) {
+    LOGGER.entering();
+    try {
+      if (INSTANCE == null) {
+        INSTANCE = new WebhookRestServer(restConfig);
+        return;
+      }
+
+      throw new IllegalStateException();
+    } finally {
+      LOGGER.exiting();
+    }
+  }
+
+  /**
+   * Accessor for obtaining reference to the RestServer singleton instance.
+   *
+   * @return RestServer - Singleton instance of the RestServer
+   */
+  public static synchronized WebhookRestServer getInstance() {
+    return INSTANCE;
+  }
+
+  /**
+   * Release RestServer singleton instance. Should only be called once. Throws IllegalStateException
+   * if singleton instance not created.
+   */
+  public static void destroy() {
+    LOGGER.entering();
+    try {
+      if (INSTANCE != null) {
+        INSTANCE = null;
+        return;
+      }
+
+      throw new IllegalStateException();
+    } finally {
+      LOGGER.exiting();
+    }
+  }
+
+  /**
+   * Defines a resource configuration that scans for JAX-RS resources and providers in the REST
+   * package.
+   *
+   * @param restConfig the operator REST configuration
+   * @return a resource configuration
+   */
+  static ResourceConfig createResourceConfig(WebhookRestConfig restConfig) {
+    ResourceConfig rc =
+        new ResourceConfig()
+            .register(JacksonFeature.class)
+            .register(ErrorFilter.class)
+            .register(RequestDebugLoggingFilter.class)
```

Comment:
I now know why your change is so big 🤣... This class is just the same as the original except that the set of filters is different (there are probably a few other differences), but I really do think the next "big thing" for this change should be removing unnecessary copies.

Reply:
Refactored to use the existing RestServer and RestConfig clases in b0ef168 .